Linkta-org/core
# Login error resolution / simplification

This PR endeavors to resolve some nagging errors and broken functionality while removing, improving, or simplifying some parts of the application.

The main issues had to do with failed database queries which trigger alerts in the application, and the user's display name failing to populate on the database after signing up with email/password.

Previously, there were two static layout components; one for authenticated users which could render "private" views in the RouterOutlet, and one for unauthenticated users which would only render "public" views in the RouterOutlet. The display of one or the other of these layouts was conditionally based on the user authentication state. In addition, there was logic in the SignUp and SignIn views which would navigate to a private view ("/generate") upon successful authentication. 

The above became problematic because there is additional logic which should run after authentication (GET or POST to database, update Firebase auth token properties). Invoking navigate() immediately upon successful authentication would unmount the SignUp or SignIn component along with its queries and unexecuted logic. Orchestrating a conditional invocation of navigate() was insufficient because the change from one static layout to the other would also cause an unmount and re-render of the child component, i.e., the SignUp or SignIn view.

The solution may be to eliminate the dual static layouts, and use any available means to delay invocation of navigate() until after all other logic has completed. 

- Eliminate the "dual static layouts" concept to prevent problematic component re-renders
- Eliminate onAuthSuccess() method from SignInPage and SignUpPage
- Migrate StaticLayout (formerly _MainLayout_) and SideNav to CSS modules
- ~Implement "find one or create" logic for user profile queries on server~
- Generally rely on Firebase-provided UID to interact with users on the database, rather than Mongo ObjectId
- Correct property access on decoded Firebase auth token
- Remove references to email and remove DB index on email
- Add Skeleton for user name in SideNav UI
- Various updates to import statements, variable names, typos, small syntax errors, etc.
- Pass trace string to useFetchUserProfile() hook to track calls from various components
- Consistent implementation of timestamps across DB models
- Delete obsolete files: genAiController.ts, typeChecker.ts, UnauthorizedLayout.tsx, UnauthorizedLayout.module.css

Fixes #356 #236 #318 #384 

## Type of change




## How Can this be tested? Testing Plan to review this PR



What are the steps to test this change?

Generally interact with the app as a user. Delete your accounts on Firebase Auth and Mongo, create new accounts, submit queries, interact with Flows, etc. Use OAuth provider buttons as well as email/password sign-up and sign-in.

Expect reliable operation without errors.


## Checklist:
